### PR TITLE
Ansible playbooks for prometheus, grafana,sensu

### PR DESCRIPTION
**Pull request by amitbasuri**
This pull request significantly streamlines and enhances the deployment and management of a Katzenpost mix network, incorporating automated infrastructure provisioning with Terraform and deployment with Ansible. Key improvements include a `Makefile` for easy command execution, detailed setup instructions in `README.md`, and the integration of monitoring tools such as Prometheus, Grafana, and Sensu for comprehensive oversight of the network's performance and health. Security group configurations have been refined for precise access control, and dynamic inventory management ensures accurate and up-to-date configuration. This PR aims to provide a seamless, automated setup experience for deploying a Katzenpost mix network with robust monitoring capabilities.

### CODE REVIEW

File: ansible/playbooks/setup_sensu.yml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      ansible.builtin.service:
+        name: sensu-agent
+        state: restarted
+
+- name: Configure Sensu checks and handlers
+  hosts: sensu_backend
+  become: true
+  vars:
+    sensu_admin_user: "admin"
+    sensu_admin_password: "P@ssw0rd!"
```
Comment (david415) [inline ansible/playbooks/setup_sensu.yml:98]: can we use an encrypted vault of some sort to handle passwords such that we aren't publish passwords in plaintext in the git repo?

File: ansible/run_playbook.sh [hunk 2]
```diff
@@ -3,14 +3,57 @@
...
+#ansible-playbook -i hosts_builder.ini playbooks/clone.yml
+
+ansible-playbook -i hosts_builder.ini playbooks/build_and_configure.yml
 
 ##### deploy mix_net #####
 ansible-playbook -i hosts_mixnet.ini playbooks/deploy_mixnet.yml
 
 #### clear artifacts ####
-rm -rf playbooks/artifacts
+#rm -rf playbooks/artifacts
```
Comment (david415) [inline ansible/run_playbook.sh:20]: remove section instead of commenting it out.

File: ansible/run_playbook.sh [hunk 3]
```diff
@@ -3,14 +3,57 @@
...
+#sudo cp  ~/voting_mixnet/auth1/link.public.pem auth1_link_pub_key.pem
+#sudo cp  ~/voting_mixnet/auth2/link.public.pem auth2_link_pub_key.pem
+#sudo cp  ~/voting_mixnet/auth3/link.public.pem auth3_link_pub_key.pem
+
+# sudo ./ping -c client.toml -s echo
+
+#ansible-galaxy collection install sensu.sensu_go
+
+## sesu Username: admin
+        #Password: P@ssw0rd!
```
Comment (david415) [inline ansible/run_playbook.sh:43]: no passwords in git

File: ansible/setup_monitoring.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
+
+export ANSIBLE_HOST_KEY_CHECKING=False
+export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES
+export CI=true
+
+#ansible-galaxy collection install prometheus.prometheus --force
```
Comment (david415) [inline ansible/setup_monitoring.sh:6]: remove uncommented section or justify it's existence

File: ansible/setup_monitoring.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+ansible-playbook -i hosts_mixnet.ini playbooks/setup_prometheus.yml
+
+ansible-playbook -i hosts_mixnet.ini playbooks/setup_grafana.yml -v
+
+ansible-playbook -i hosts_mixnet.ini playbooks/setup_grafana_dashboard.yml -v
+
+ansible-playbook -i hosts_mixnet.ini playbooks/setup_sensu.yml -vv
+
+#grafana
```
Comment (david415) [inline ansible/setup_monitoring.sh:19]: remove uncommented section or explain why it should exist

File: ansible/setup_monitoring.sh [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+#http://54.166.205.206:3000/
+#Username: admin Password: admin
```
Comment (david415) [inline ansible/setup_monitoring.sh:21]: no passwords in git

**Review by david415 — CHANGES_REQUESTED** [review]
Looks great! There's just a few minor things to fix.

File: ansible/infra/sg.tf [hunk 7]
```diff
@@ -0,0 +1,139 @@
+resource "aws_security_group" "ssh_access" {
+  name        = "ssh_access"
+  description = "Allow SSH from anywhere"
+
+  ingress {
+    from_port   = 22
```
Comment (david415) [inline ansible/infra/sg.tf:6]: Can we setup sshd to listen on a random high port instead of 22? such as 33429

**Comment by david415** [comment]
We want a publicly accessible URL that we can use to navigate to the grafana prometheus dashboard AND the sensu status page. However we want this to be private... which i think means htdigest password protection

File: ansible/playbooks/deploy_mixnet.yml [hunk 8]
```diff
@@ -7,13 +7,14 @@
...
+  hosts: dirauth_nodes:provider_nodes:mix_nodes
   become: true
   strategy: free
-  tasks:
-    - name: Include common setup tasks
-      include_tasks: roles/common/tasks/main.yml
+  roles:
+    - common
+    - chrony
+    - node_exporter
```
Comment (xendarboh) [inline ansible/playbooks/deploy_mixnet.yml:17]:
```
ERROR! the role 'node_exporter' was not found in /home/.../ansible/playbooks/roles:/home/user/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/home/.../ansible/playbooks

The error appears to be in '/home/.../ansible/playbooks/deploy_mixnet.yml': line 17, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

    - chrony
    - node_exporter
      ^ here
```

Comment (amitbasuri) [reply ansible/playbooks/deploy_mixnet.yml]: @xendarboh updated the code, we don't need this, we will use setup_prometheus.yml to setup node exporter.

File: ansible/infra/sg.tf (hunk 7 above)
Comment (amitbasuri) [reply ansible/infra/sg.tf]: @david415 yes should be possible, how about doing this in a different PR?

File: ansible/playbooks/deploy_mixnet.yml (hunk 8 above)
Comment (amitbasuri) [reply ansible/playbooks/deploy_mixnet.yml]: plz check new README for usage

**Comment by amitbasuri** [comment]
> We want a publicly accessible URL that we can use to navigate to the grafana prometheus dashboard AND the sensu status page. However we want this to be private... which i think means htdigest password protection

sure, how about doing this as part of a different PR, also added README for better usage.

File: ansible/run_playbook.sh (hunk 2 above)
Comment (david415) [reply ansible/run_playbook.sh:20]: Please please please please please remove commented out section of code.

File: ansible/run_playbook.sh [hunk 9]
```diff
@@ -3,14 +3,57 @@
...
+
+#sudo apt-get update
```
Comment (david415) [inline ansible/run_playbook.sh:45]: remove section of commented out code. this is what git is for.

Comment (amitbasuri) [reply ansible/run_playbook.sh]: removed

File: ansible/infra/sg.tf (hunk 7 above)
Comment (david415) [reply ansible/infra/sg.tf]: ok

**Review by david415 — APPROVED** [review]

**Review by xendarboh — APPROVED** [review]

**Pull request merged**